### DISCUSSION TITLE
Camera Focusable

### DISCUSSION THREAD

**Discussion by waikeat2397** [raspberrypi/picamera2#706] [Q&A] [answered]
Is there any API that may return a value state that whether the camera has a motorised lens or not? Read through the documentation and seems like now there aren't any API to do that. Try to do some application that can fit with any type of camera supported by raspberry pi. Just that I can't identify whether the camera has a motorised lens programatically until I start the autofocus cycle, then only it will prompt out that the camera does not support autofocus.

**Comment by davidplowman** [comment, marked as answer]
I think the AfMode property should only be in the camera controls (`picam2.camera_controls`) if there is a focusable lens available.
